pkg/components/power-monitor/deployment.go

```diff
@@ -545,6 +547,55 @@ func NewPowerMonitorUWMTokenSecret(d components.Detail, pmi *v1alpha1.PowerMonit
 	}
 }
 
+// AnnotateServiceMonitorWithSecretHash sets annotations on the ServiceMonitor to trigger a rollout when the Secret changes
+func AnnotateServiceMonitorWithSecretHash(sm *monv1.ServiceMonitor, s *corev1.Secret) {
```

Comment:
Can we instead of `ServiceMonitor` use an interface that allows getting and setting annotation?